tests/integration/models/reddit/test_rules.py

```diff
@@ -0,0 +1,194 @@
+from unittest import mock
+import pytest
+
+from praw.exceptions import RedditAPIException, ClientException
+from praw.models import Rule
+
+from ... import IntegrationTest
+
+
+class TestRule(IntegrationTest):
+    @property
+    def subreddit(self):
+        return self.reddit.subreddit(pytest.placeholders.test_subreddit)
+
+    def test_add_rule(self):
+        self.reddit.read_only = False
+        with self.recorder.use_cassette("TestRule.test_add_rule"):
+            rule = self.subreddit.rules.mod.add(
+                "PRAW Test",
+                "all",
+                description="Test by PRAW",
+                violation_reason="PTest",
+            )
+            assert rule.short_name == "PRAW Test"
+            assert rule.kind == "all"
+            assert rule.description == "Test by PRAW"
+            assert rule.violation_reason == "PTest"
+
+    def test_add_rule_without_violation_reason(self):
+        self.reddit.read_only = False
+        with self.recorder.use_cassette(
+            "TestRule.test_add_rule_without_violation_reason"
+        ):
+            rule = self.subreddit.rules.mod.add("PRAW Test 2", "comment")
+            assert rule.short_name == "PRAW Test 2"
+            assert rule.kind == "comment"
+            assert rule.description == ""
+            assert rule.violation_reason == "PRAW Test 2"
+
+    @mock.patch("time.sleep", return_value=None)
+    def test_delete_rule(self, _):
+        self.reddit.read_only = False
+        with self.recorder.use_cassette("TestRule.test_delete_rule"):
+            rules = list(self.subreddit.rules)
+            rule = rules[-1]
+            rule.mod.delete()
+            assert len(list(self.subreddit.rules)) == (len(rules) - 1)
+
+    def test_iter_rules(self):
+        with self.recorder.use_cassette("TestRule.test_iter_rules"):
+            for rule in self.subreddit.rules:
+                assert isinstance(rule, Rule)
+
+    @pytest.mark.filterwarnings("ignore", category=DeprecationWarning)
+    def test_iter_call(self):
+        with self.recorder.use_cassette("TestRule.test_iter_rules"):
+            for rule in self.subreddit.rules():
+                assert isinstance(rule, Rule)
+
+    def test_iter_rule_string(self):
+        with self.recorder.use_cassette("TestRule.test_iter_rules"):
+            rule = self.subreddit.rules["PRAW Test"]
+            assert isinstance(rule, Rule)
+            rule._fetch()
+            assert "kind" in vars(rule)
```

Comment:
Couldn't this just be `assert rule.kind`?

Reply:
That is true.